app/controllers/users/edit_mobile_controller.rb

```diff
@@ -0,0 +1,39 @@
+module Users
+  class EditMobileController < ApplicationController
+    include PhoneConfirmation
+
+    before_action :confirm_two_factor_authenticated
```

Comment:
I just think we should be careful that the auth actions always happen first.. just in case one of the other actions happens to perform an action as the user (who might not be 2fa'd).  Unlikely I know.